src/cpp/utilities/refine.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#ifndef __UTILS_REFINE
+#define __UTILS_REFINE
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+
+#include <fstream>
+
+namespace UtilsMesh {
+
+    using namespace dealii;
+
+
+    /* @brief metafunction for refinement of a mesh up to n steps of refinement.
+
+     @param triangulation the mesh into a triangulation
+     @param steps the amount of refinement steps
+     @param save_into where the mesh is to be saved as a .vtu file
```

Comment:
This will not be a VTU file, but an MSH file. "No documentation is better than wrong documentation"